crates/wasm-smith/src/code_builder.rs

```diff
@@ -508,18 +508,23 @@ where
         let mut instructions = vec![];
 
         while !self.allocs.controls.is_empty() {
-            let keep_going =
-                instructions.len() < max_instructions && u.arbitrary().unwrap_or(false);
+            let keep_going = instructions.len() < max_instructions
+                && u.arbitrary().map_or(false, |b: u8| b != 0);
             if !keep_going {
                 self.end_active_control_frames(&mut instructions);
                 break;
             }
 
             gather_options(module, &mut self);
-
-            let f = u.choose(&self.allocs.options)?;
-            let inst = f(u, module, &mut self)?;
-            instructions.push(inst);
+            if let Ok(f) = u.choose(&self.allocs.options) {
```

Comment:
Let's change this from an `if let` into a `match` like

```rust
match u.choose(..) {
    Ok(f) => {
        // existing stuff from if let
    }
    Err(_) => {
        self.end_active_control_frames(&mut instructions);
        break;
    }
}
```

This is functionally the same, since the next iteration of the loop will have `keep_going == false` because the `u.arbitrary()` will return an `Err`, but I think better conveys intent and desired control flow.